Execute scripts on page navigation (view transitions) (#7786)
withastro/astro: .changeset/sharp-seas-smile.md

---
'astro': patch
---

> Execute scripts when navigating to a new page
> 
> When navigating to an new page with client-side navigation, scripts are executed (and re-executed). This makes it so that any new scripts on the incoming page are run and DOM can be updated.
> 
> Module scripts won't re-execute because they never do. To support cases where you want to modify the DOM this also adds a `astro:load` event that you can listen to:

Reviewer: ```suggestion
Execute scripts when navigating to a new page.

When navigating to an new page with client-side navigation, scripts are executed (and re-executed) so that any new scripts on the incoming page are run and the DOM can be updated.

However, `type=module` scripts never re-execute in Astro, and will not do so in client-side routing. To support cases where you want to modify the DOM, a new `astro:load` event listener been added:
```
Some tiny grammar tidying, plus some suggestions for concise wording (if accurate!) for your consideration!
